Library/Homebrew/migrator.rb

```diff
@@ -147,47 +169,51 @@ def pinned?
   end
 
   def migrate
-    if old_cellar.exist? && new_cellar.exist?
+    oh1 "Migrating #{Formatter.identifier(oldname)} to #{Formatter.identifier(newname)}"
+    lock
+    unlink_oldname
+    unlink_newname if new_cellar.exist?
+    repin
+    move_to_new_directory
+    link_oldname_cellar
+    link_oldname_opt
+    link_newname unless old_linked_keg.nil?
```

Comment:
What if both `newname` and `oldname` were linked? Here we'll link migrated `oldname` keg if any.